Storage/DiskImageStore.swift

```diff
@@ -44,50 +43,85 @@ open class DiskImageStore {
     }
 
     /// Gets an image for the given key if it is in the store.
-    open func get(_ key: String) -> Deferred<Maybe<UIImage>> {
-        return deferDispatchAsync(queue) { () -> Deferred<Maybe<UIImage>> in
-            if !self.keys.contains(key) {
-                return deferMaybe(DiskImageStoreErrorType(description: "Image key not found"))
-            }
+    open func getImageForKey(_ key: String) async throws -> UIImage {
+        return try await withCheckedThrowingContinuation { continuation in
+            queue.async {
+                if !self.keys.contains(key) {
+                    continuation.resume(throwing: DiskImageStoreErrorCase.notFound(description: "Image key not found"))
+                    return
+                }
 
-            let imagePath = URL(fileURLWithPath: self.filesDir).appendingPathComponent(key)
-            if let data = try? Data(contentsOf: imagePath),
-               let image = UIImage.imageFromDataThreadSafe(data) {
-                return deferMaybe(image)
+                let imagePath = URL(fileURLWithPath: self.filesDir).appendingPathComponent(key)
+                do {
+                    let data = try Data(contentsOf: imagePath)
+                    if let image = UIImage.imageFromDataThreadSafe(data) {
+                        continuation.resume(returning: image)
+                    } else {
+                        continuation.resume(throwing: DiskImageStoreErrorCase.invalidImageData(description: "Invalid image data"))
+                    }
+                } catch {
+                    continuation.resume(throwing: error)
+                }
             }
-
-            return deferMaybe(DiskImageStoreErrorType(description: "Invalid image data"))
         }
     }
 
     /// Adds an image for the given key.
     /// This put is asynchronous; the image is not recorded in the cache until the write completes.
     /// Does nothing if this key already exists in the store.
-    @discardableResult open func put(_ key: String, image: UIImage) -> Success {
-        return deferDispatchAsync(queue) { () -> Success in
-            let imageURL = URL(fileURLWithPath: self.filesDir).appendingPathComponent(key)
-            if let data = image.jpegData(compressionQuality: self.quality) {
-                do {
-                    try data.write(to: imageURL, options: .noFileProtection)
-                    self.keys.insert(key)
-                    return succeed()
-                } catch {
-                    self.logger.log("Unable to write image to disk: \(error)",
-                                    level: .warning,
-                                    category: .storage)
+    @discardableResult
+    open func saveImageForKey(_ key: String, image: UIImage) async throws -> Success {
```

Comment:
We want to move away from using the Success type as it's part of Deferred. It looks like the previous put function never uses this return type anyway so you can probably just remove it.
I don't think this function needs to throw either as nobody is observing it in the current system and I think we will mostly likely continue to fail silently in the new system too, as we have a viable fall back in the case no image is available.

Reply:
I missed that block when doing the changes because the name is so generic, thank you for pointing it out.